### PR TITLE
[#4605] Handle NPE in Doris table operation prop append

### PR DESCRIPTION
**Pull request by itsaviu**
### What changes were proposed in this pull request?

In `DorisTableOperations`, handled a possible Null Pointer Exception scenario 

### Why are the changes needed?

Fix: (#4605)

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Local Test Run - Passed

### CODE REVIEW

File: catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java [hunk 1]
```diff
@@ -170,13 +170,12 @@ protected String generateCreateTableSql(
...
   private Map<String, String> appendNecessaryProperties(Map<String, String> properties) {
-    Map<String, String> resultMap;
     if (properties == null) {
-      resultMap = new HashMap<>();
-    } else {
-      resultMap = new HashMap<>(properties);
+      return new HashMap<>();
     }
 
+    Map<String, String> resultMap = new HashMap<>(properties);
```
Comment (yuqi1129) [inline catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java:177]: I believe replacing `properties` with `resultMap` in L184 can solve the problem, your changes I think can't address it.

Comment (itsaviu) [reply catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java:177]: My thoughts was, the problem in L184 `!properties.containsKey(REPLICATION_FACTOR` would be `properties` can be null and NPE could occur, so in L173 `properties == null` we check if it is NULL and return a empty Map right way.

Comment (itsaviu) [reply catalogs/catalog-jdbc-doris/src/main/java/org/apache/gravitino/catalog/doris/operation/DorisTableOperations.java]: @yuqi1129 , have made the changes, will resolve the convo after you check.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**